pkg/install/consolebranding.go

```diff
@@ -0,0 +1,75 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+func (i *Installer) updateConsoleBranding(ctx context.Context) error {
+	restConfig, err := restconfig.RestConfig(ctx, i.env, i.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	cli, err := operatorclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	i.log.Print("waiting for console-operator config")
+	timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Minute)
+	defer cancel()
+	err = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		_, err := cli.OperatorV1().Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		return err == nil, nil
```

Comment:
I see that `PollImmediateUntil` returns `err` if it is not `nil`. I'm curious about why the `error` returned here will always be `nil`. It will return `true, nil` if there is no error, and `false, nil` if there is one; `err != nil` on line 39 would always be `false`.

Reply:
The rationale is that it ignores transitory errors (API server misbehaving because it's not ready, or restarting, or something) and keeps trying until the 30 minutes is up or we succeed.
On line 39 err will be non-nil (context expired) if we time out - this comes from within PollImmediateUntil.

Reply:
Makes sense. Thanks for the explanation.